File: app/src/main/java/com/saifi369/databindingexample/MainActivity.java
package com.saifi369.databindingexample;

import android.os.Bundle;

import com.saifi369.databindingexample.databinding.MainActivityBinding;
import com.saifi369.databindingexample.viewmodels.MainViewModel;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;

public class MainActivity extends AppCompatActivity {

    private MainViewModel mViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        MainActivityBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        mViewModel = ViewModelProviders.of(this)
                .get(MainViewModel.class);

        binding.setMainViewModel(mViewModel);

    }
}

File: app/src/main/java/com/saifi369/databindingexample/MainActivity.kt
package com.saifi369.databindingexample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction()
                .add(R.id.main_fragment_container, MainFragment(), "main_fragment")
                .commit()
    }
}